views-pebble/src/main/java/io/micronaut/views/pebble/PebbleLoader.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.views.pebble;
+
+import com.mitchellbosecke.pebble.loader.ClasspathLoader;
+import io.micronaut.views.ViewUtils;
+import io.micronaut.views.ViewsConfiguration;
+
+/**
+ * Loader for Pebble templates.
+ *
+ * @author Ecmel Ercan
+ * @since 3.0.2
+ */
+public class PebbleLoader extends ClasspathLoader {
+
+    private final String extension;
+
+    public PebbleLoader(ViewsConfiguration views, PebbleConfiguration pebble){
+        super.setPrefix(ViewUtils.normalizeFolder(views.getFolder()));
+        extension = ViewUtils.EXTENSION_SEPARATOR + pebble.getDefaultExtension();
+    }
+
+    private String normalizeTemplateName(String templateName) {
```

Comment:
Annotate the parameter String templateName with `@NonNull` 

Annotate the return type with `@NonNull`

Reply:
Done